features/FEATURE_LWIP/lwip-interface/lwip-eth/arch/TARGET_NXP/TARGET_LPCTarget/lpc17_emac.c

```diff
@@ -611,7 +611,9 @@ static err_t lpc_low_level_output(struct netif *netif, struct pbuf *p)
 	/* Wait until enough descriptors are available for the transfer. */
 	/* THIS WILL BLOCK UNTIL THERE ARE ENOUGH DESCRIPTORS AVAILABLE */
 #if NO_SYS == 0
-	for (idx = 0; idx < dn; idx++) {
+	/* It is safe to cast dn to uint32 here since it is either derived from the
+	 * uint16 returned by pbuf_clen() above, or it is set to 1. */
+	for (idx = 0; idx < (u32_t) dn; idx++) {
```

Comment:
If we can avoid a cast, would be sweet. In this case the problem is that this unsigned `idx` is intended for use elsewhere as the unsigned descriptor index, but this loop is actually comparing with the descriptor count.

No reason to use the same variable. Just have a local `s_32_t count` as the loop variable here.